boards/tenstorrent/tt_blackhole/tt_blackhole_fixed_partitions.dtsi

```diff
@@ -14,101 +20,214 @@ partitions {
+	cmfwcfg: partition@1f7000 {
+		label = "cmfwcfg";
+		reg = <0x1f7000 DT_SIZE_K(4)>;
+		binary-path = "$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/fw_table.bin";
 		read-only;
 	};
 
-	ethfw: partition@31000 {
-		label = "ethfw";
-		reg = <0x31000 DT_SIZE_K(64)>;
-		binary-path = "$BLOBS_DIR/tt_blackhole_erisc.bin";
+
+	ethfwcfg: partition@1f8000 {
+		label = "ethfwcfg";
+		reg = <0x1f8000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_erisc_params.bin";
 		read-only;
 	};
 
-	memfwcfg: partition@41000 {
+	memfwcfg: partition@1f9000 {
 		label = "memfwcfg";
-		reg = <0x41000 DT_SIZE_K(4)>;
+		reg = <0x1f9000 DT_SIZE_K(4)>;
 		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_params_$BOARD_REV.bin";
 		read-only;
 	};
 
-	memfw: partition@42000 {
-		label = "memfw";
-		reg = <0x42000 DT_SIZE_K(16)>;
-		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_init.bin";
+	ethsdreg: partition@1fa000 {
+		label = "ethsdreg";
+		reg = <0x1fa000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fwreg.bin";
 		read-only;
 	};
 
-	ethsdreg: partition@46000 {
-		label = "ethsdreg";
-		reg = <0x46000 DT_SIZE_K(4)>;
-		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fwreg.bin";
+	flshinfo: partition@1fb000 {
+		label = "flshinfo";
+		reg = <0x1fb000 DT_SIZE_K(4)>;
+		binary-path =
+			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
+		read-only;
+	};
+
+	/* Remote chip firmware images */
+	ethfw: partition@1fc000 {
+		label = "ethfw";
+		reg = <0x1fc000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_erisc.bin";
+		read-only;
+	};
+
+	memfw: partition@20c000 {
+		label = "memfw";
+		reg = <0x20c000 DT_SIZE_K(64)>;
+		binary-path = "$BLOBS_DIR/tt_blackhole_gddr_init.bin";
 		read-only;
 	};
 
-	ethsdfw: partition@47000 {
+	ethsdfw: partition@21c000 {
 		label = "ethsdfw";
-		reg = <0x47000 DT_SIZE_K(20)>;
+		reg = <0x21c000 DT_SIZE_K(64)>;
 		binary-path = "$BLOBS_DIR/tt_blackhole_serdes_eth_fw.bin";
 		read-only;
 	};
 
-	bmfw: partition@4c000 {
-		label = "bmfw";
-		reg = <0x4c000 DT_SIZE_K(68)>;
+	/* Update firmware image for the DMC, as seen by mcuboot */
+	dmfw: partition@22d000 {
+		label = "dmfw";
+		reg = <0x22d000 DT_SIZE_K(452)>;
+	};
+
+	/*
+	 * Partition where image is actually written. Offset by 4KB to
+	 * support swap using move.
+	 */
+	dmfwimg: partition@22e000 {
+		label = "dmfwimg";
+		reg = <0x22e000 DT_SIZE_K(448)>;
 		binary-path = "$BUILD_DIR/dmc/zephyr/zephyr.signed.bin";
-		read-only;
 	};
 
-	flshinfo: partition@59000 {
-		label = "flshinfo";
-		reg = <0x59000 DT_SIZE_K(4)>;
-		binary-path =
-			"$BUILD_DIR/smc/generated_board_cfg/$BOARD_REV/flash_info.bin";
-		read-only;
+	/*
+	 * Image trailer for the DMC firmware update. Indicates to MCUBoot that
+	 * the image is valid and should be booted in test mode.
+	 */
+	dmfw_tail: partition@29d000 {
+		label = "dmfwtail";
+		reg = <0x29d000 DT_SIZE_K(4)>;
+		binary-path = "$BLOBS_DIR/mcuboot_magic_test.bin";
 	};
```

Comment:
It's probably worth it to mention here that the trailer is normally part of the slot / mcuboot image, and that this trailer partition is not needed for regular firmware updates.

https://docs.mcuboot.com/design.html#image-trailer

From what I understand, this is only needed for the BL2 update process.

I personally lean on "self-documenting code", rather than heavily-commented code, because the latter can suffer from bit rot, which often ends up being a source of confusion.